src/argilla/cli/server/search_engine/reindex.py

```diff
@@ -0,0 +1,113 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import AsyncGenerator
+
+from rich.progress import Progress
+from sqlalchemy import func, select
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+
+from argilla.cli import typer_ext
+from argilla.server.database import AsyncSessionLocal
+from argilla.server.models import Dataset, Record, Response, Suggestion
+from argilla.server.search_engine import SearchEngine, get_search_engine
+
+
+class Reindexer:
+    YIELD_PER = 100
+
+    @classmethod
+    async def reindex_datasets(cls, db: AsyncSession, search_engine: SearchEngine) -> AsyncGenerator[Dataset, None]:
+        stream = await db.stream(
+            select(Dataset)
+            .order_by(Dataset.inserted_at.asc())
+            .options(
+                selectinload(Dataset.fields),
+                selectinload(Dataset.questions),
+                selectinload(Dataset.metadata_properties),
+                selectinload(Dataset.vectors_settings),
+            )
+            .execution_options(yield_per=cls.YIELD_PER)
+        )
+
+        async for dataset in stream.scalars():
+            await search_engine.delete_index(dataset)
+            await search_engine.create_index(dataset)
+
+            yield dataset
+
+    @classmethod
+    async def reindex_dataset_records(
+        cls, db: AsyncSession, search_engine: SearchEngine, dataset: Dataset
+    ) -> AsyncGenerator[Record, None]:
+        stream = await db.stream(
+            select(Record)
+            .filter_by(dataset_id=dataset.id)
+            .order_by(Record.inserted_at.asc())
+            .options(
+                selectinload(Record.responses).selectinload(Response.user),
+                selectinload(Record.suggestions).selectinload(Suggestion.question),
+                selectinload(Record.vectors),
+            )
+            .execution_options(yield_per=cls.YIELD_PER)
+        )
+
+        async for record in stream.scalars():
+            await search_engine.index_records(record.dataset, [record])
+
+            yield record
+
+    @classmethod
+    async def count_datasets(cls, db: AsyncSession) -> int:
+        return (await db.execute(select(func.count(Dataset.id)))).scalar()
+
+    @classmethod
+    async def count_dataset_records(cls, db: AsyncSession, dataset: Dataset) -> int:
+        return (await db.execute(select(func.count(Record.id)).filter_by(dataset_id=dataset.id))).scalar()
+
+
+async def _reindex_datasets(db: AsyncSession, search_engine: SearchEngine, progress: Progress) -> None:
+    task = progress.add_task(
+        f"reindexing datasets...",
```

Comment:
Here you can specify that only the Feedback datasets will be reindexed. Just to clarify to users

Reply:
Done.